src/main/scala/io/iohk/ethereum/consensus/ethash/difficulty/ConfiguredDifficultyCalculator.scala

```diff
@@ -0,0 +1,34 @@
+package io.iohk.ethereum.consensus.ethash.difficulty
+
+import io.iohk.ethereum.consensus.difficulty.DifficultyCalculator
+import io.iohk.ethereum.domain.BlockHeader
+
+class ConfiguredDifficultyCalculator(powTargetTime: Long) extends DifficultyCalculator {
+
+  import EthashDifficultyCalculator._
+
+  /**
+    * The LowerBoundExpectedRatio (l for abbreviation below) divides the timestamp diff into ranges:
+    *   [0, l)   => c = 1, difficulty increases
+    *   [l, 2*l) => c = 0. difficulty stays the same
+    *   ...
+    *   [l*i, l*(i+1) ) => c = 1-i, difficulty decreases
+    *
+    *  example:
+    *  powTargetTime := 45 seconds
+    *  l := 30 seconds
+    *   [0, 0.5 min)         => difficulty increases
+    *   [0.5 min, 1 min)   => difficulty stays the same (the average should be powTargetTime)
+    *   [1 min, +infinity) => difficulty decreases
+    */
+  private val LowerBoundExpectedRatio: Long = (powTargetTime / 1.5).toLong
+
+  def calculateDifficulty(blockNumber: BigInt, blockTimestamp: Long, parentHeader: BlockHeader): BigInt = {
+    val timestampDiff = blockTimestamp - parentHeader.unixTimestamp
+
+    val x: BigInt = parentHeader.difficulty / DifficultyBoundDivision
+    val c: BigInt = math.max(1 - (timestampDiff / LowerBoundExpectedRatio), FrontierTimestampDiffLimit)
+
+    MinimumDifficulty.max(parentHeader.difficulty + x * c)
```

Comment:
How did we arrive at this formula? I would have expected it to be parameterised by something like:
- mu - the expected block time
- sigma - std deviation, where we don't change the diff
- delta - rate of change outside of std deviation range

Perhaps something like: https://en.wikipedia.org/wiki/Beta_distribution could be used. The formula we have is eerily similar ETH formula, but I think we are allowed more freedom with this, no?

Disclaimer: these are very rough proposals and are not to be taken word-for-word. But I think something along those lines would give us more control. It's not a blocker if such control is not required.

Reply:
This was inspired by what I had done on this regard on the other project so I'll share my thoughts. As you said this formula is the ETH formula with simplifications, I didn't want to deviate too much from it just in case I'd break something (for example, if we are more aggressive with when to increase/decrease difficulty it might lead to an divergent time between blocks).

Either way, the only reason I see for maintaining the current formula is so that this doesn't become a blocker for deploying checkpointing on our testnet, if that isn't at risk we could go for another better one

(of course, if this formula doesn't work as expected we should definitely change it!)

Reply:
I would go now with the current formula for three reasons: 1. we need this fast for testnet deployment 2. It is the same formula used in other project and its upgraded version is in mantis now. 3. Developing new formula only for testnet without having in mind using it in mainnet is not worth a time now imho.

Maybe we can think about it in the future? (new task)

Reply:
I would try it out on the testnet and only create a follow-up task if we detect there that's not enough (or that it would be good to have something better)